api/routes/api_login.py

```diff
@@ -95,8 +96,21 @@ def reset_password(token, service: UserService = Provide[Container.user_service]
 async def log_in(service: UserService = Provide[Container.user_service]):
     try:
         data = request.get_json()
-        dto = InputUserLoginDTO().load(data)
-        user = await service.log_in(dto.email_or_username, dto.password_hash)
+        dto = T
+        status = ""
+
+        try:
+            dto = InputUserByGoogleDTO().load(data) or dto
+            status = "google"
```

Comment:
Hardcoded string which will have to be moved to some predefined space (like Enum)